### DISCUSSION TITLE
Omitting determiners from merged noun chunks

### DISCUSSION THREAD

**Discussion by lolfert** [explosion/spaCy#10743] [Help: Coding & Implementations] [answered]
Is there a way to omit determiners when merging noun phrases when using the `merge_noun_chunks` pipeline function? Right now I'm using a custom pipeline component and [Textacy's noun chunking function](https://textacy.readthedocs.io/en/latest/api_reference/extract.html#textacy.extract.basics.noun_chunks) to accomplish this, but if there's a way to do this using only spaCy I'd rather do that.

To clarify, I'd for instance like `The quick brown fox jumps over the lazy dog` to be tokenized as `The [quick brown fox] jumps over the [lazy dog]` instead of `[The quick brown fox] jumps over [the lazy dog]`.

**Comment by rmitsch** [comment, marked as answer]
Hi @lolfert! Customized noun chunk merging is out of the box not possible with `merge_noun_chunks()`.

A spaCy-only alternative to using Textacy is a custom pipeline component post-processing noun chunks with a modified implementation of [`merge_noun_chunks()`](https://github.com/explosion/spaCy/blob/master/spacy/pipeline/functions.py#L17). If you look up the [source for Textacy's noun_chunks()](https://textacy.readthedocs.io/en/latest/_modules/textacy/extract/basics.html#noun_chunks), you'll see it's rather straightforward (same with `merge_noun_chunks()`). It should be easy enough to adapt those for your purposes and write a solution achieving something similar using only spaCy.

**Reply by lolfert** [reply]
That's what I figured, thanks for the help!
